### ISSUE TITLE
Refactor and Purpose of lightning-net-tokio

### ISSUE THREAD

**Issue by ariard** [rust-bitcoin/rust-lightning#423]
What the long-term purpose of lightning-net-tokio ? Is it to provide a default networking stack to avoid people writing tricky asynchronous code and in this case is it worthy to maintain ?

We were wondering with @valentinewallace as part of refactoring it to support newer versions of tokio and futures.

**Comment by TheBlueMatt** [comment]
Right, kinda unclear. Having a “import + type one line” way of having a network stack is pretty important imo. Whether it should integrate with tokio...I don’t really know. I think I’d be ok with rewriting it to just have one thread and use mio?

**Comment by ariard** [comment]
Okay, I might try to rewrite it, I need something for testing purpose, but only once we are done with other refactorings
